### PR TITLE
[MySQL Compatibility 2/4][Bug] Fix bug and improve compatibility with mysql protocol

### LINKED ISSUE #4361: [Bug][MySQL compatibility] Query to info schema table does not return correct results.
**Describe the bug**
```
SELECT TABLE_SCHEMA TABLE_CAT, NULL  TABLE_SCHEM,  TABLE_NAME, IF(TABLE_TYPE='BASE TABLE', 'TABLE', TABLE_TYPE) as TABLE_TYPE, TABLE_COMMENT REMARKS, NULL TYPE_CAT, NULL TYPE_SCHEM, NULL TYPE_NAME, NULL SELF_REFERENCING_COL_NAME,  NULL REF_GENERATION FROM INFORMATION_SCHEMA.TABLES  WHERE (ISNULL(database()) OR (TABLE_SCHEMA = database())) AND (TABLE_NAME LIKE '%') AND TABLE_TYPE IN ('BASE TABLE','VIEW','FOREIGN TABLE','MATERIALIZED VIEW','EXTERNAL TABLE') ORDER BY TABLE_TYPE, TABLE_SCHEMA, TABLE_NAME;
```

Above sql will return empty, which expected to return the table info in the current database.

**Why**

This is because `database()` function will return the full name of a database like `default_cluster:db1`.
But `TABLE_SCHEMA` returned from `INFORMATION_SCHEMA.TABLES` is the database name without `default_cluster`.
So the where predicate `TABLE_SCHEMA = database()` is false.

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

1. select database() will only return database name, without cluster name. #4361
    
2. select user() will return the IP which user connected in. #4360

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] I have create an issue on (Fix #4361 #4360 ), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/InformationFunction.java [hunk 1]
```diff
@@ -75,3 +80,5 @@ public String toSqlImpl() {
         return funcType + "()";
     }
 }
+
+
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/InformationFunction.java:84]: two empty lines？

Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/InformationFunction.java:84]:
```suggestion
```

**Review by kangkaisen — DISMISSED** [review]
+1

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**